### PR TITLE
Update API based on latest spec

### PR DESCRIPTION
**Pull request by wlan0**
- remove anonymous access mode from bucket and bucketclass
 - refactor package

### CODE REVIEW

**Comment by wlan0** [comment]
this addresses #29 and #24

File: apis/objectstorage.k8s.io/v1alpha1/types.go [hunk 1]
```diff
@@ -21,63 +21,81 @@ import (
...
+	SchemeBuilder.Register(&BucketAccessRequest{}, &BucketAccessRequestList{})
+	SchemeBuilder.Register(&BucketAccessClass{}, &BucketAccessClassList{})
 }
 
-type AnonymousAccessMode struct {
-	// +optional
-	Private bool `json:"private,omitempty"`
-	// +optional
-	PublicReadOnly bool `json:"publicReadOnly,omitempty"`
+type RetentionPolicy string
```
Comment (jeffvance) [inline apis/objectstorage.k8s.io/v1alpha1/types.go:34]: The lastest kep pr defines this field as "deletionPolicy" based on review meeting consensus.

File: apis/objectstorage.k8s.io/v1alpha1/types.go [hunk 2]
```diff
@@ -21,63 +21,81 @@ import (
...
+
+const (
+	RetentionPolicyRetain RetentionPolicy = "Retain"
+	RetentionPolicyDelete RetentionPolicy = "Delete"
```
Comment (jeffvance) [inline apis/objectstorage.k8s.io/v1alpha1/types.go:38]: and we've added a 3rd value: "ForceDelete"

File: apis/objectstorage.k8s.io/v1alpha1/types.go (hunk 1 above)
Comment (wlan0) [reply apis/objectstorage.k8s.io/v1alpha1/types.go:34]: Good catch. I’ll update it

**Comment by jeffvance** [comment]
@wlan0 I'd like to see KEP pr [2349](https://github.com/kubernetes/enhancements/pull/2349/) reviewed and agreed to before reviewing this pr, unless you're okay with making additonal changes once the kep pr is merged?

**Comment by wlan0** [comment]
@jeffvance fair point. While I’m happy to make changes here, let’s get the kep reviewed

**Pull request merged**